### PR TITLE
Add code for store and customer

### PR DESCRIPTION
**Pull request by royfalk**
- Fix bug in resource
- Add tests for resource and store

I have added unit tests instead of the standard tests as the code has not yet been integrated with engine.
I am waiting for @stephengtuggy 's changes, as he's touching the same areas of the code.

Code Changes:
- [ ] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation No. 
- [ ] This is a documentation change only

Issues:
- None

Purpose:
- What is this pull request trying to do? Make the code cleaner.
- What release is this for?
- Is there a project or milestone we should apply this to?

### CODE REVIEW

File: engine/src/resource/store.cpp [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    }
+
+    return -1;
+}
+
+
+
+bool Store::Sell(Customer& buyer, std::string product_name, double quantity)
+{
+    buyer.Buy(*this, product_name, quantity);
```
Comment (BenjamenMeyer) [inline engine/src/resource/store.cpp:119]: missing return value (thus the GH check failures)

**Review by BenjamenMeyer — CHANGES_REQUESTED** [review]
I like the direction this is going

File: engine/src/resource/store.cpp (hunk 1 above)
Comment (royfalk) [reply engine/src/resource/store.cpp]: Yeah, I noticed that. Weird though that **_my_** IDE and compiler were fine with it.

Comment (royfalk) [reply engine/src/resource/store.cpp]: Are you good with my changes? I need an approval please.

**Review by BenjamenMeyer — APPROVED** [review]

**Review by stephengtuggy — APPROVED** [review]

**Pull request merged**